### DISCUSSION TITLE
Does Koreader overwrite CSS styling embedded in epub files?

### DISCUSSION THREAD

**Discussion by fritzrehde** [koreader/koreader#11113] [Q&A] [answered]
Hi, I am using a Kobo Libra H2O and have always had the problem that some epub files, which I store in calibre, have fixed line height, and margin width settings embedded in the CSS. This means that with the native Kobo reader, I can't adjust those parameters at all. My (hacky) solution has been: manually remove the CSS styling I don't want (i.e. line height and margin width) using Calibre's editor. However, this is somewhat cumbersome, and I think it would be useful if whatever reader I use can override any CSS stylings from the epub. So my question is: Does the Koreader reader support overwriting some of these CSS styling settings? Would I run into the same issue as with the native Kobo reader where I can't adjust line height or margin width at all?
Thanks!

**Comment by NiLuJe** [comment, marked as answer]
Why yes, we do. Enter the magical world of style tweaks.

Feel free to peruse the [user guide](http://koreader.rocks/user_guide/) for context.

**Reply by fritzrehde** [reply]
Awesome!

**Reply by poire-z** [reply]
You can use our own set of generic style tweaks we make available in our top menu to achieve most of what you want:
https://github.com/koreader/koreader/blob/master/frontend/ui/data/css_tweaks.lua

If that's not enough (or their effect is too radical) with some books, you can select some text in the book and tap View HTML in the menu to... view the HTML, the full stylesheets, and also when tapping on some html elements, the CSS rules that do match that element (and its parents). Some screenshots in https://github.com/koreader/crengine/pull/511 .
You can then set your own CSS rules to couteract them for this book with the Book style tweak menu item, which launch a text editor (where you need to use `!important` in them to override the book stylesheets rules).
